includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -681,6 +692,11 @@ public function authorize_user() {
 			exit();
 		}
 
+		$current_user_id = get_current_user_id();
+		if ( $this->is_owner_must_be_changed( $this->owner_id->get(), $current_user_id ) ) {
+			$this->owner_id->set( $current_user_id );
+		}
```

Comment:
The placement here seems to be a bit arbitrary, and separates a check for the access token and the method that sets the access token which make sense to keep together. Let's move this down between the calls to `set_granted_scopes` and `refresh_profile_data`.

Also, we need to add a test for this to `OAuth_ClientTest`. Since the `test_authorize_user` method is already quite large, let's add a test for the new `should_update_owner_id` method directly. Ideally we wouldn't do that because it's private, but in this case I think it's actually cleaner.

Reply:
Updated.